src/services/blockchain/__tests__/eth-bc-service.test.ts

```diff
@@ -93,11 +98,63 @@ describe('ETH service connected to ganache', () => {
       ).toNumber()
     ).toEqual(firstRetry.toNumber())
     expect(
-      EthereumBlockchainService.increaseGasPricePerAttempt(gasEstimate.maxPriorityFeePerGas, 2, firstRetry).toNumber()
+      EthereumBlockchainService.increaseGasPricePerAttempt(
+        gasEstimate.maxPriorityFeePerGas,
+        2,
+        firstRetry
+      ).toNumber()
     ).toEqual(secondRetry.toNumber())
   })
 })
 
+describe('setGasPrice', () => {
+  const cid = new CID('bafyreic5p7grucmzx363ayxgoywb6d4qf5zjxgbqjixpkokbf5jtmdj5ni')
+  const feeData = {
+    maxFeePerGas: BigNumber.from(2000),
+    maxPriorityFeePerGas: BigNumber.from(1000),
+    gasPrice: BigNumber.from(1000),
+  }
+  const gasLimit = BigNumber.from(10)
+  const provider = {
+    estimateGas: jest.fn(() => gasLimit),
+    getNetwork: jest.fn(() => ({ chainId: '1337' })),
+    getTransactionCount: jest.fn(),
+    getFeeData: jest.fn(() => feeData),
+  }
+
+  const buildBlockchainService = async (provider: any): Promise<EthereumBlockchainService> => {
+    const wallet = {
+      address: 'abcd1234',
+      provider: provider,
+      sendTransaction: jest.fn(),
+    }
+    const ethBc = new EthereumBlockchainService(config, wallet as any)
+    await ethBc.connect()
+    return ethBc
+  }
+
+  test('legacy transaction', async () => {
+    const legacyProvider = Object.assign({}, provider, {
+      getFeeData: jest.fn(() => ({ gasPrice: feeData.gasPrice })),
+    })
+    const ethBc = await buildBlockchainService(legacyProvider)
+    const txData = await ethBc._buildTransactionRequest(cid)
+    for (const attempt of [0, 1, 2]) {
+      await ethBc.setGasPrice(txData, attempt)
+      expect(txData).toMatchSnapshot()
```

Comment:
sigh.  this test doesn't really accomplish what I was hoping for.  snapshot tests should be used very sparingly IMO as they don't really convey any semantic value to the reader about what they are trying to test.  They'll tell us if the response changes, but not in a way that lets future people making changes to this code understand whether that change is good or bad.  I was hoping for a test that specifically targeted reproducing the issue that we saw in our environment where some anchors were failing due to being underpriced on a transaction retry because the gas price had fallen but transaction retries always have to be priced higher than previous attempts.  That way we would be specifically testing the condition that lead to that underpricing happening, and have a test that is clearly testing for transaction underpricing due to retry logic and changing gas fees.  That way even if the exact gas pricing logic changes in the future, we'd still have a test case that forces the author of the new gas price logic (who might not be you or I) to remember to consider this scenario.

In any case, it's not important enough to keep holding up this review any longer, but something to keep in mind for the future.